lib/Lift/InstructionTranslator.cpp

```diff
@@ -1143,56 +1088,43 @@ IT::translateOpcode(PTCOpcode Opcode,
 
     return v{ Builder.CreateOr(FirstShift, SecondShift) };
   }
-  case PTC_INSTRUCTION_op_deposit_i32:
-  case PTC_INSTRUCTION_op_deposit_i64: {
-    unsigned Position = ConstArguments[0];
-    if (Position == RegisterSize)
-      return v{ InArguments[0] };
-
-    unsigned Length = ConstArguments[1];
-    uint64_t Bits = 0;
-
-    // Thou shall not << 32
-    if (Length == RegisterSize)
-      Bits = getMaxValue(RegisterSize);
-    else
-      Bits = (1 << Length) - 1;
-
-    // result = (t1 & ~(bits << position)) | ((t2 & bits) << position)
-    uint64_t BaseMask = ~(Bits << Position);
-    Value *MaskedBase = Builder.CreateAnd(InArguments[0], BaseMask);
-    Value *Deposit = Builder.CreateAnd(InArguments[1], Bits);
-    Value *ShiftedDeposit = Builder.CreateShl(Deposit, Position);
-    Value *Result = Builder.CreateOr(MaskedBase, ShiftedDeposit);
-
+  case LIBTCG_op_deposit_i32:
+  case LIBTCG_op_deposit_i64: {
+    Value *Result = genDeposit(Builder, RegisterSize,
+                               InArguments[0], InArguments[1],
+                               InArguments[2], InArguments[3]);
```

Comment:
Just outlining or functional changes?

Reply:
Only outlining, we need genDeposit to implement extract2 in a later commit